### PR TITLE
executables_db: avoid `Formula#all`

### LINKED ISSUE #133: Using brew gives warnings
Warning: Calling Formula#all without `--all` or `HOMEBREW_EVAL_ALL` is deprecated! There is no replacement.
Please report this issue to the homebrew/command-not-found tap (not Homebrew/brew or Homebrew/homebrew-core):
```
/opt/homebrew/Library/Taps/homebrew/homebrew-command-not-found/lib/executables_db.rb:57
```

### PR DESCRIPTION
**Pull request by woodruffw**
Instead, collect all formulae from their taps, then load them explicitly via `Formulary`.

This should fix #133.

### CODE REVIEW

**Comment by woodruffw** [comment]
Not stale.

File: lib/executables_db.rb [hunk 1]
```diff
@@ -54,40 +56,45 @@ def changed?
     def update!(update_existing: false, install_missing: false, max_downloads: nil)
       downloads = 0
       disabled_formulae = []
-      Formula.all.each do |f|
-        break if max_downloads.present? && downloads > max_downloads.to_i
-        next if f.tap?
 
-        name = f.full_name
+      Tap.each do |tap|
```
Comment (MikeMcQuaid) [inline lib/executables_db.rb:60]: @woodruffw Probably want to limit this to official taps only without `--eval-all` or just require `--eval-all` or `HOMEBREW_EVAL_ALL` to be passed to the relevant commands?

For context: the concern here is arbitrary Ruby execution from untrusted taps/formulae.

**Review by MikeMcQuaid — COMMENTED** [review]
Apologies for the long delay here. Have watched this repo.

File: lib/executables_db.rb (hunk 1 above)
Comment (woodruffw) [reply lib/executables_db.rb:60]: Yeah, good call -- I'll have some time to update here in the coming days 🙂

Comment (MikeMcQuaid) [reply lib/executables_db.rb:60]: @woodruffw Gentle nudge

Comment (woodruffw) [reply lib/executables_db.rb:60]: Thanks for the nudge, looking again today 😅

Comment (woodruffw) [reply lib/executables_db.rb:62]: `brew which-update` now takes `--eval-all`, which is disabled by default (meaning that, by default, only `CoreTap.instance` is used).

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks again @woodruffw!

**Pull request merged**